Automattic/wp-calypso
# composite-checkout: Replace coupon display with "Total savings"

#### Changes proposed in this Pull Request

This modifies the review step in composite checkout to display "Total savings" (with a sublabel of any coupon code) instead of "Coupon". This is because the amount displayed by that line item is actually the total savings from all discounts, not just coupons.

Since savings may exist without a coupon, this also displays the "Total savings" line item any time there's a discount in the cart.

Finally, it corrects some inconsistencies about when we display "billed annually" for a line item. The logic there was unclear previously and likely had bugs.

**NOTE** This exposes some bugs around how the total savings is calculated and rendered. These bugs will need to be fixed on the server, probably before we merge this PR. D44528-code 
 and D44573-code should fix those issues.

#### Screenshots

Before:
![before](https://user-images.githubusercontent.com/2036909/83931433-bc8dc100-a76a-11ea-8fc0-7fc57144abd2.png)

After:

![after](https://user-images.githubusercontent.com/2036909/83931437-c0214800-a76a-11ea-8479-a0e25526d23e.png)

#### Testing instructions

- Visit composite checkout with a plan in the cart; verify that you see the price with no savings line item displayed.
- Add a domain to your cart as well as a plan; verify that you see the savings line item include the total saved for the bundled domain.
- Add a coupon to your cart; verify that you see the savings from the coupon displayed in the savings line item as well as the coupon code itself.
